### ISSUE TITLE
Unrelated no member named 'fbegin'  ?

### ISSUE THREAD

**Issue by diorcety** [serge-sans-paille/pythran#814]
I can't understand why a compilation error is triggered with the following example

```
import numpy as np

def bb(data):
    out_data = np.zeros((10, ))
    hist = out_data[:-1]
    return hist

def aa(data):
    _data = data.flatten()
    return bb(_data)

def cc(data):
    return bb(data)

#pythran export test()
def test():
    a = np.zeros((10, ))
    b = np.zeros((10, ))

    c = cc(a)  # replace cc by aa for triggering the issue
    d = cc(b)  # replace cc by aa for triggering the issue
    c[d == 0] = np.nan
```

**Comment by serge-sans-paille** [comment]
I cannot reproduce :-/

**Comment by diorcety** [comment]
Here the logs: https://pastebin.com/tEPC77nR

**Comment by serge-sans-paille** [comment]
Got it, thanks!

**Comment by diorcety** [comment]
`data[data == 0] = np.nan` seems to crash also ...

**Comment by diorcety** [comment]
```
c = data == 0
data[c] = np.nan
```
produce an other issue

**Comment by serge-sans-paille** [comment]
I believe this is fixed by #929, can you confirm?

**Comment by serge-sans-paille** [comment]
@diorcety can you confirm it's ok on master?
